src/error.rs

```diff
@@ -19,6 +21,66 @@ pub enum ProtocolError {
     UnsupportedCompressionFlag(u8),
 }
 
+impl<T> fmt::Display for Error<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Grpc(ref _status, ref _header_map) =>
+                write!(f, "gRPC error"),
+            Error::Protocol(ref _protocol_error) =>
+                write!(f, "Protocol error"),
+            Error::Decode(ref _decode_error) =>
+                write!(f, "Message decode error"),
+            Error::Inner(ref _inner) =>
+                write!(f, "Inner error"),
+        }
+    }
+}
+
+impl fmt::Display for ProtocolError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ProtocolError::MissingTrailers =>
+                write!(f, "Missing trailers"),
+            ProtocolError::MissingMessage =>
+                write!(f, "Missing message"),
+            ProtocolError::UnexpectedEof =>
+                write!(f, "Unexpected EOF"),
+            ProtocolError::Internal =>
+                write!(f, "Internal"),
+            ProtocolError::UnsupportedCompressionFlag(flag) =>
+                write!(f, "Unsupported compression flag: {}", flag),
+        }
+    }
+}
+
+impl std::error::Error for ProtocolError {
+    fn description(&self) -> &str {
```

Comment:
Nit: `Error::description` is deprecated now, it's suggested to leave this unimplemented and just use `fmt::Display` instead.